src/KeyRegistry.sol

```diff
@@ -422,6 +452,16 @@ contract KeyRegistry is TrustedCaller, Signatures, EIP712, Nonces {
         KeyData storage keyData = keys[fid][key];
         if (keyData.state != KeyState.NULL) revert InvalidState();
 
+        /* First byte of metadata must encode type ID */
+        uint8 typeId = uint8(metadata[0]);
```

Comment:
the type encoded as the first byte starts to feel like its adding complexity. should we: 

1. require the type as an additional param? 
2. merge the type with the scheme so that they are one and the same?

Reply:
I dislike option 2 because it means we can't add more than one metadata type for a scheme. (Maybe an OK tradeoff in practice). 1 is probably cleaner, I'll try it.